src/wwwroot/controllers/MainCtrl.js

```diff
@@ -63,7 +65,22 @@
         isFreeTrialAlmostEnded : false,
         isFreeTrialEndToday : false
       }
-      
+      if(!auth.getFreeTrialNotificationFromStorage() && !$rootScope.modalOpened) {
```

Comment:
You should compare stored date with current date in order to show again the modal if the trial was extended and expired again.